cmd/state-svc/test/integration/svc_int_test.go

```diff
@@ -0,0 +1,119 @@
+package integration
+
+import (
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/shirou/gopsutil/process"
+	"github.com/stretchr/testify/suite"
+)
+
+type SvcIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *SvcIntegrationTestSuite) TestStartStop() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("start"))
+	cp.Expect("Starting")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Checking")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("stop"))
+	cp.Expect("Stopping")
+	cp.ExpectExitCode(0)
+}
+
+func (suite *SvcIntegrationTestSuite) TestSignals() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	// SIGINT (^C)
+	cp := ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("foreground"))
+	cp.Expect("Starting")
+	cp.Signal(syscall.SIGINT)
+	cp.Expect("caught a signal: interrupt")
+	cp.ExpectNotExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+
+	// SIGTERM
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("foreground"))
+	cp.Expect("Starting")
+	cp.Signal(syscall.SIGTERM)
+	suite.NotContains(cp.TrimmedSnapshot(), "caught a signal")
+	cp.ExpectNotExitCode(0)
+
+	cp = ts.SpawnCmdWithOpts(ts.SvcExe, e2e.WithArgs("status"))
+	cp.Expect("Service cannot be reached")
+	cp.ExpectExitCode(1)
+}
+
+func (suite *SvcIntegrationTestSuite) TestSingleSvc() {
+	suite.OnlyRunForTags(tagsuite.Service)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	oldCount := suite.GetNumStateSvcProcesses() // may be non-zero due to non-test state-svc processes
```

Comment:
I think we can make this more certain. Maybe add a function to svcctl that checks the sock file directory and sends stop to each socket before any tests run. To segregate the service instances running in tests versus the system running the tests, the temp directory envvar can be set (I think that should work) so that the sock file directory is set/known.

Reply:
Is that really necessary in the context of running unit tests on GitHub Actions? I only added this for running tests on my local machine where I already have state-svc processes running. On GitHub Actions I don't think there should really be any.

Reply:
Then I suggest splitting the non-ci code out in block using `e2e.RunningOnCI()` so that it's more clear that it is not meant to be helpful there.

Reply:
Hmm, I was wrong. It looks like there is a running, non-test state-svc process on CI.

Reply:
I think it's the installer subshell.

Reply:
We run tests with the state tool, so yeah we can expect one state-svc process to be running that's unrelated to our tests.